gprofiler/profilers/java.py

```diff
@@ -768,6 +872,19 @@ def _init_java_safemode(self, java_safemode: str) -> None:
                 f" --java-safemode={JavaSafemodeOptions.JAVA_EXTENDED_VERSION_CHECKS}"
             )
 
+    def _init_java_flag_retrieval(self, java_jvm_flags_to_retrieve: str) -> Union[JavaFlagRetrievalOptions, List[str]]:
+        if java_jvm_flags_to_retrieve is None or java_jvm_flags_to_retrieve == "":
```

Comment:
`is None`? Type is defined as `: str`.

Reply:
I want to support `None` as default for example when called from `make_java_profiler`, although I pass default value explicitly, if you want Ill remove it

Reply:
This code is confusing for the reader.

If the API allows None - then you can check for None and specify it in the APi.
If the API doesn't allow None - don't check for it, assert it. Or don't do anything and trust mypy. But at most, assert.

assert is read differently than ifs

Reply:
Fixed already removed the check for None